src/main/java/uk/gov/pay/adminusers/service/InviteRouter.java

```diff
@@ -19,13 +20,23 @@ public InviteRouter(InviteServiceFactory inviteServiceFactory, InviteDao inviteD
         this.inviteDao = inviteDao;
     }
 
-    public Optional<Pair<InviteCompleter, Boolean>> routeComplete(String inviteCode) {
+    public Optional<InviteCompleter> routeComplete(String inviteCode) {
         return routeIfExist(inviteCode,
                 inviteEntity -> {
-                    boolean isServiceType = inviteEntity.isServiceType();
-                    InviteCompleter inviteCompleter = isServiceType ? inviteServiceFactory.completeServiceInvite() : inviteServiceFactory.completeUserInvite();
-                    return Optional.of(Pair.of(inviteCompleter, isServiceType));
-                });
+                    switch (inviteEntity.getType()) {
+                        case SERVICE:
+                        case NEW_USER_AND_NEW_SERVICE_SELF_SIGNUP:
+                            return Optional.of(Pair.of(inviteServiceFactory.completeSelfSignupInvite(), true));
+                        case USER:
+                        case EXISTING_USER_INVITED_TO_EXISTING_SERVICE:
+                            return Optional.of(Pair.of(inviteServiceFactory.completeExistingUserInvite(), false));
+                        case NEW_USER_INVITED_TO_EXISTING_SERVICE:
+                            return Optional.of(Pair.of(inviteServiceFactory.completeNewUserExistingServiceInvite(), true));
+                        default:
+                            throw new WebApplicationException(String.format("Unrecognised invite type: %s", inviteEntity.getType()));
+                    }
+                })
+                .map(Pair::getLeft);
```

Comment:
If we're calling `getLeft` does the switch statement need to return a Pair at all? Are we using the right value?

Reply:
You're right; I've now rationalised this and squashed it into the existing rationalisation commit.